File: ems-frontend/src/services/authService.js
import API from "../api/axiosConfig";

export const login = (data) => {
    return API.post("/auth/login", data);
};

export const register = (data) => {
    return API.post("/auth/register", data);
};

export const changePassword = (data) => {
    return API.put("/auth/change-password", data);
};
export const forgotPassword = (email) => {
    return API.post("/auth/forgot-password", {
        email
    });
};

export const verifyOtp = (email, otp) => {
    return API.post("/auth/verify-otp", {
        email,
        otp
    });
};

export const resetPassword = (email, otp, newPassword) => {
    return API.post("/auth/reset-password", {
        email,
        otp,
        newPassword
    });
};

File: EmployeeManagementSystem/src/main/java/org/mahesh/ems/service/AuthService.java
package org.mahesh.ems.service;

import org.mahesh.ems.dto.ChangePasswordRequest;
import org.mahesh.ems.dto.request.LoginRequest;
import org.mahesh.ems.dto.request.RegisterRequest;

public interface AuthService {

    String register(RegisterRequest request);
    String login(LoginRequest request);
    void changePassword(String email, ChangePasswordRequest request);
    void forgotPassword(String email);

    void verifyOtp(String email, String otp);

    void resetPassword(
            String email,
            String otp,
            String newPassword
    );

}
